### PR TITLE
Add a problem matcher for gcc warnings/errors: for develop commits, and branch of PR

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - @Myoldmopar requested it on https://github.com/NREL/EnergyPlus/pull/10827#issuecomment-2497204431

Pwned from https://github.com/microsoft/vscode-cpptools/blob/bfa3c7544ef63e7e0b6d0bd6bf8a1363ed3c4deb/Extension/package.json#L439

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: .github/workflows/cpp-problem-matcher.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+          "file": 1,
+          "line": 2,
+          "column": 3,
+          "severity": 4,
+          "message": 5
+        }
+      ]
+    }
+  ]
+}
```
Comment (jmarrec) [inline .github/workflows/cpp-problem-matcher.json:17]: cf https://regex101.com/r/4mPjR3/1 for demo

File: .github/workflows/test_develop_commits.yml [hunk 2]
```diff
@@ -135,7 +135,10 @@ jobs:
     - name: Build
       id: build
       working-directory: ./build
-      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+      run: |
+        echo "::add-matcher::.github/workflows/cpp-problem-matcher.json"
+        cmake --build . -j ${{ matrix.nproc }} --config Release
+        echo "::remove-matcher owner=gcc-problem-matcher::"
```
Comment (jmarrec) [inline .github/workflows/test_develop_commits.yml]: building develop commits, we add it, build, and remove it.

File: .github/workflows/test_pull_requests.yml [hunk 3]
```diff
@@ -155,7 +155,10 @@ jobs:
     - name: Branch Build
       id: branch_build
       working-directory: ./branch/build
-      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+      run: |
+        echo "::add-matcher::.github/workflows/cpp-problem-matcher.json"
+        cmake --build . -j ${{ matrix.nproc }} --config Release
+        echo "::remove-matcher owner=gcc-problem-matcher::"
```
Comment (jmarrec) [inline .github/workflows/test_pull_requests.yml]: Testing PR: enable it only when building the PR branch (not develop for regression)

File: .github/workflows/test_pull_requests.yml [hunk 4]
```diff
@@ -155,7 +155,10 @@ jobs:
     - name: Branch Build
       id: branch_build
       working-directory: ./branch/build
-      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+      run: |
+        echo "::add-matcher::.github/workflows/cpp-problem-matcher.json"
```
Comment (jmarrec) [inline .github/workflows/test_pull_requests.yml]:
```suggestion
        echo "::add-matcher::../../.github/workflows/cpp-problem-matcher.json"
```

File: .github/workflows/test_develop_commits.yml [hunk 5]
```diff
@@ -135,7 +135,10 @@ jobs:
     - name: Build
       id: build
       working-directory: ./build
-      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+      run: |
+        echo "::add-matcher::.github/workflows/cpp-problem-matcher.json"
```
Comment (jmarrec) [inline .github/workflows/test_develop_commits.yml]:
```suggestion
        echo "::add-matcher::../.github/workflows/cpp-problem-matcher.json"
```

**Comment by Myoldmopar** [comment]
Nailed it!  OK, so the add-matcher command seems to ignore working-directory.  Good to know.  Thanks @jmarrec this was _fun_ and will be very helpful.

**Pull request merged**